File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let max = 0;
    let buy = 0;
    let sell = 1;

    while (sell < prices.length) {
        let profit = prices[sell] - prices[buy];
        max = Math.max(max, profit);
        if (prices[sell] < prices[buy]) buy = sell;
        sell++;
    }
    return max;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = 0
        max_profit = 0
        for sell in range(len(prices)):
            print('sell=', sell, 'buy=', buy)
            profit = prices[sell] - prices[buy]
            max_profit = max(profit, max_profit)
            if prices[sell] < prices[buy]:
                buy = sell
        return max_profit